code/ship/ship.cpp

```diff
@@ -1768,77 +1768,112 @@ ship_info::~ship_info()
 	free_strings();
 }
 
+static SCP_vector<SCP_string> Removed_ships;
+
 /**
  * Parse the information for a specific ship type.
  */
 static void parse_ship(const char *filename, bool replace)
 {
-	char buf[SHIP_MULTITEXT_LENGTH];
+	char fname[NAME_LENGTH];
 	ship_info *sip = nullptr;
-	bool create_if_not_found  = true;
+	bool first_time = false;
+	bool create_if_not_found = true;
+	bool remove_ship = false;
 
 	required_string("$Name:");
-	stuff_string(buf, F_NAME, SHIP_MULTITEXT_LENGTH);
+	stuff_string(fname, F_NAME, NAME_LENGTH);
+	diag_printf("Ship name -- %s\n", fname);
 
-	if(optional_string("+nocreate")) {
-		if(!replace) {
+	if (optional_string("+nocreate")) {
+		if (!replace) {
 			Warning(LOCATION, "+nocreate flag used for ship in non-modular table");
 		}
 		create_if_not_found = false;
 	}
 
+	// check first if this is on the remove blacklist
+	auto it = std::find(Removed_ships.begin(), Removed_ships.end(), fname);
+	if (it != Removed_ships.end()) {
+		remove_ship = true;
+	}
```

Comment:
you could simplify this with `remove_ship = std::any_of(Removed_ships.begin(), Removed_ships.end(), [](const auto& it) { return it == fname;});`

Reply:
I wouldn't say that's "simpler". :)  I prefer to keep it the way it is to be consistent with the rest of the code here.